### ISSUE TITLE
Error building Commit 41d59fd23b20af8493726acbb9f7257728d82d6d

### ISSUE THREAD

**Issue by ShinduHub** [rlguy/Blender-FLIP-Fluids#367]
#### System Information

**Blender Version (including hash):**2.79 hash: f4dc9f9d68b
**FLIP Fluids Version:**Branch: Master Commit: 41d59fd23b20af8493726acbb9f7257728d82d6d
**Operating System:**Arch Linux
**CPU:** Intel Core I7 7700HQ (Notebook)
**GFX:** Nvidia Geforce GTX 1050 (Notebook)
**RAM:** 8GB
(It does not matter for this issue, because the issue is in the code.)

#### Describe the bug

Error while Building the latest version.

ErrorLog:
```
[5/45] Building CXX object CMakeFiles/objects.dir/src/engine/clscalarfield.cpp.o
FAILED: CMakeFiles/objects.dir/src/engine/clscalarfield.cpp.o 
/bin/g++  -DCL_USE_DEPRECATED_OPENCL_1_2_APIS -DWITH_OPENCL=1 -I../src -O3 -std=c++11 -Wall -fPIC -pthread -O3 -DNDEBUG -MD -MT CMakeFiles/objects.dir/src/engine/clscalarfield.cpp.o -MF CMakeFiles/objects.dir/src/engine/clscalarfield.cpp.o.d -o CMakeFiles/objects.dir/src/engine/clscalarfield.cpp.o -c ../src/engine/clscalarfield.cpp
In file included from ../src/engine/clscalarfield.h:30,
                 from ../src/engine/clscalarfield.cpp:25:
../src/engine/arrayview3d.h: In Elementfunktion »void ArrayView3d<T>::getViewAsArray3d(Array3d<T>&)«:
../src/engine/arrayview3d.h:193:16: Fehler: Return-Anweisung mit Wert, in Funktion mit Rückgabetyp »void« [-fpermissive]
         return view;
                ^~~~
[10/45] Building CXX object CMakeFiles/objects.dir/src/engine/fluidsimulation.cpp.o
FAILED: CMakeFiles/objects.dir/src/engine/fluidsimulation.cpp.o 
/bin/g++  -DCL_USE_DEPRECATED_OPENCL_1_2_APIS -DWITH_OPENCL=1 -I../src -O3 -std=c++11 -Wall -fPIC -pthread -O3 -DNDEBUG -MD -MT CMakeFiles/objects.dir/src/engine/fluidsimulation.cpp.o -MF CMakeFiles/objects.dir/src/engine/fluidsimulation.cpp.o.d -o CMakeFiles/objects.dir/src/engine/fluidsimulation.cpp.o -c ../src/engine/fluidsimulation.cpp
In file included from ../src/engine/clscalarfield.h:30,
                 from ../src/engine/fluidsimulation.h:44,
                 from ../src/engine/fluidsimulation.cpp:25:
../src/engine/arrayview3d.h: In Elementfunktion »void ArrayView3d<T>::getViewAsArray3d(Array3d<T>&)«:
../src/engine/arrayview3d.h:193:16: Fehler: Return-Anweisung mit Wert, in Funktion mit Rückgabetyp »void« [-fpermissive]
         return view;
                ^~~~
[11/45] Building CXX object CMakeFiles/objects.dir/src/engine/main.cpp.o
FAILED: CMakeFiles/objects.dir/src/engine/main.cpp.o 
/bin/g++  -DCL_USE_DEPRECATED_OPENCL_1_2_APIS -DWITH_OPENCL=1 -I../src -O3 -std=c++11 -Wall -fPIC -pthread -O3 -DNDEBUG -MD -MT CMakeFiles/objects.dir/src/engine/main.cpp.o -MF CMakeFiles/objects.dir/src/engine/main.cpp.o.d -o CMakeFiles/objects.dir/src/engine/main.cpp.o -c ../src/engine/main.cpp
In file included from ../src/engine/clscalarfield.h:30,
                 from ../src/engine/fluidsimulation.h:44,
                 from ../src/engine/main.cpp:27:
../src/engine/arrayview3d.h: In Elementfunktion »void ArrayView3d<T>::getViewAsArray3d(Array3d<T>&)«:
../src/engine/arrayview3d.h:193:16: Fehler: Return-Anweisung mit Wert, in Funktion mit Rückgabetyp »void« [-fpermissive]
         return view;
                ^~~~
[14/45] Building CXX object CMakeFiles/objects.dir/src/engine/diffuseparticlesimulation.cpp.o
ninja: build stopped: subcommand failed.
```

#### To Reproduce

Terminal:
`git clone https://github.com/rlguy/Blender-FLIP-Fluids && cd Blender-FLIP-Fluids && mkdir bin && cd bin && cmake -GNinja ../ && cmake --build .`
(I prefer googles Ninja as my Build system)

#### Expected Behaviour

Build succeed.

#### Actual Behaviour

Build failed.

#### Screenshots
![bildschirmfoto von 2018-08-02 21-12-31](https://user-images.githubusercontent.com/26170289/43605539-f66cefa2-9698-11e8-909d-cc9b9d4b57ca.png)

**Comment by ShinduHub** [comment]
It is easily fixable with replacing the void return statements with Array3d<T>

**Comment by rlguy** [comment]
Hey ShinduHub,

Thanks for letting me know about this! That function shouldn't actually return anything so the return should just be deleted. It looks like this function isn't used in the engine, so that is why my compiler did not give me an error.

This fix will be included in the next version of the addon (**1.0.4**). Note that this repository issue tracker is for the Blender Market version of the product, so it may take some time until this fix is included in the repository code. (See [CONTRIBUTING.md](https://github.com/rlguy/Blender-FLIP-Fluids/blob/master/CONTRIBUTING.md))

\- Ryan
